File: examples/openai_client.mjs
import OpenAI from 'openai'

const client = new OpenAI({
  baseURL: process.env.DGX_OPENAI_BASE_URL ?? 'http://dgx-spark.local:3000/v1',
  apiKey: process.env.DGX_OPENAI_API_KEY,
})

const stream = await client.chat.completions.create({
  model: process.env.DGX_MODEL ?? 'qwen3.8-27b',
  messages: [{ role: 'user', content: 'Describe the DGX Spark in one sentence.' }],
  stream: true,
})

for await (const chunk of stream) {
  process.stdout.write(chunk.choices[0]?.delta?.content ?? '')
}
process.stdout.write('\n')

File: examples/openai_client.py
import os

from openai import OpenAI

client = OpenAI(
    base_url=os.getenv("DGX_OPENAI_BASE_URL", "http://dgx-spark.local:3000/v1"),
    api_key=os.environ["DGX_OPENAI_API_KEY"],
)

for chunk in client.chat.completions.create(
    model=os.getenv("DGX_MODEL", "qwen3.8-27b"),
    messages=[{"role": "user", "content": "Describe the DGX Spark in one sentence."}],
    stream=True,
):
    print(chunk.choices[0].delta.content or "", end="", flush=True)
print()
